Source/RapyutaSimulationPlugins/Private/Robots/RRRobotROS2Interface.cpp

```diff
@@ -67,7 +69,8 @@ bool URRRobotROS2Interface::InitPublishers()
             OdomPublisher->bPublishOdomTf = bPublishOdomTf;
         }
         OdomPublisher->InitializeWithROS2(RobotROS2Node);
-        OdomPublisher->RobotVehicle = Robot;
+        // If publishing odom, it must be an [ARRRobotBaseVehicle]
+        OdomPublisher->RobotVehicle = CastChecked<ARRRobotBaseVehicle>(Robot);

```

Comment:
Same question as above : can this make the editor crash if `Robot` is not properly initialized?

Reply:
Yes. `CastChecked` acts like a type check for the owner `Robot` if turning on `bPublishOdom`